### PR TITLE
Fix/proxy

### PR DESCRIPTION
**Pull request by tejohnso**
@ahmedalsudani please review

### CODE REVIEW

File: network.js [hunk 1]
```diff
@@ -43,16 +43,15 @@ function setJavaProxyArgs(fields) {
     `-Dhttp.proxyHost=${fields.hostname}`,
     `-Dhttp.proxyPort=${fields.port}`,
     `-Dhttps.proxyHost=${fields.hostname}`,
-    `-Dhttps.proxyPort=${fields.port}`
+    `-Dhttps.proxyPort=${fields.port}`,
+    `-Dhttp.nonProxyHosts="localhost|127.0.0.1|0.0.0.0"`
```
Comment (ahmedalsudani) [inline network.js]: @tejohnso shouldn't we also exclude other local IP ranges like `10.*` as we do on the Electron side?

File: proxy.js [hunk 2]
```diff
@@ -1,28 +1,63 @@
 var urlParse = require("url").parse,
-observers = [];
+platform = require("./platform.js"),
+observers = [],
+path = require("path"),
+pacScriptPath = path.join(platform.getInstallDir(), "proxy-pac.js"),
```
Comment (ahmedalsudani) [inline proxy.js]: @tejohnso so this will be in ~/rvplayer?

File: proxy.js [hunk 3]
```diff
@@ -1,28 +1,63 @@
...
-      configObj.address = "http://" + configObj.address;
+    if (configObj.username) {
+      configObj.auth = `${configObj.username}`;
+      if (configObj.password) {
+        configObj.auth += `:${configObj.password}`;
+      }
+    } else {
+      if (configObj.auth) {
+        configObj.username = configObj.auth.split(":")[0];
+        configObj.password = configObj.auth.split(":")[1];
```
Comment (ahmedalsudani) [inline proxy.js]: @tejohnso What if the password includes a colon?

File: network.js (hunk 1 above)
Comment (tejohnso) [inline network.js]: @ahmedalsudani Looking more closely now, I think we might be better off to leave the nonProxyHosts param out https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html

File: proxy.js (hunk 2 above)
Comment (tejohnso) [inline proxy.js]: Yeah, easy to find.

File: proxy.js (hunk 3 above)
Comment (tejohnso) [inline proxy.js]: @ahmedalsudani good catch - updated

File: network.js (hunk 1 above)
Comment (ahmedalsudani) [inline network.js]: @tejohnso why would we leave it out?

Comment (tejohnso) [inline network.js]: "The default value excludes all common variations of the loopback address."

Comment (ahmedalsudani) [inline network.js]: @tejohnso but that doesn't include addresses in local networks (for example 192.168.1.1). I guess the question is should we exclude those addresses or not. We are excluding them in the pac file, but not here.

Comment (tejohnso) [inline network.js]: @ahmedalsudani updated

**Comment by ahmedalsudani** [comment]
:+1:

**Pull request merged**